### PR TITLE
docs: add blog post for js-ipfs 48

### PR DESCRIPTION
**Pull request by achingbrain**
Adds post for the js-IPFS release today

### CODE REVIEW

File: content/post/104-js-ipfs-0.48.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+JS IPFS has traditionally primarily targeted the browser, and the browser is a bad place to be if you want to be on the [DHT][]. You typically aren't on a page long enough to make or respond to [DHT][] queries and nor are you diallable so even if you were able to advertise yourself as a provider for a given block, the chances are no-one can connect to you to retrieve that block which results in a degraded service for everyone. Worse, the way you find more peers and content is via the [DHT][] so you're kind of stuck.
+
+There are several ways to give in-browser IPFS nodes a better experience on the network, one of those is [Delegate Nodes](https://blog.ipfs.io/2019-08-06-js-ipfs-0-37/#delegated-peer-and-content-routing). A Delegate Node is a network peer that performs certain actions on behalf of other nodes on the network. In this case it will make [DHT][] queries on our behalf so we can find more peers and more content than ever before.
+
+`js-IPFS@0.48.0` enables delegate nodes in the configuration by default. That means you should see far more peers that you have previously and be able to find content faster and more reliably than ever before.
+
```
Comment (vasco-santos) [inline content/post/104-js-ipfs-0.48.md:22]: We should mention that the default config uses public delegate nodes, but in a production environment other delegate nodes should be setup to guarantee availability and avoid congestion. I have in my TODO list to create some documentation around libp2p infrastructure/deployment guidelines [libp2p/js-libp2p#696](https://github.com/libp2p/js-libp2p/issues/696) that will be useful for this. But meanwhile, I think we can just say that people should configure their own as we already have some information on how to do that, but it is spread across multiple channels

Comment (achingbrain) [reply content/post/104-js-ipfs-0.48.md:22]: Done!

File: content/post/104-js-ipfs-0.48.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+Gateway (read only) listening on /ip4/127.0.0.1/tcp/9090/http
+Web UI available at http://127.0.0.1:5002/webui
+Daemon is ready
+```
+
+When used as a [module](https://github.com/ipfs/js-ipfs/blob/1760b8928dac14b3abcfa4a889042f0d7a956386/packages/ipfs/docs/MODULE.md) in the browser or an application [options.repoAutoMigrate](https://github.com/ipfs/js-ipfs/blob/1760b8928dac14b3abcfa4a889042f0d7a956386/packages/ipfs/docs/MODULE.md#optionsrepoautomigrate) is enabled by default so the upgrade will happen invisibly in the background; all you will notice is a one-off slightly longer startup time.
+
+## 🗺️ A more intuitive API
+
+As the IPFS ecosystem grows more and more developers become interested in the project and start using our APIs. A lot of them have grown organically over time and not all of them have had equal amounts of thought put into them.
```
Comment (jacobheun) [inline content/post/104-js-ipfs-0.48.md:98]:
```suggestion
As the IPFS ecosystem grows more and more developers become interested in the project and start using our APIs. A lot of them have grown organically over time and not all of them have had equal amounts of time invested in them.
```

File: content/post/104-js-ipfs-0.48.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+In that last piece we lost a little bit of usability, as the humble `ipfs.add` API takes all manner of arguments and always returns an AsyncIterator:
+
+```javascript
+for await (const file of ipfs.add('My file content')) {
+  // Wait, what?  I only added one file, why am I in a loop?
+}
+```
+
+With `js-IPFS@0.48.0`, `ipfs.add` now returns a single item. This seemingly innocuous change brings a raft of usability improvements as a very common question is 'I added a file, and got back `[object AsyncGenerator]`, what is that?' and then you have to get the whiteboard and the pens out and before you know it you've gone down an long explanation cul de sac when all they wanted to do was get a [CID][] back.
```
Comment (jacobheun) [inline content/post/104-js-ipfs-0.48.md:116]:
```suggestion
With `js-IPFS@0.48.0`, `ipfs.add` now returns a single item. This seemingly innocuous change brings a raft of usability improvements as a very common question is 'I added a file, and got back `[object AsyncGenerator]`, what is that?' and then you have to get the whiteboard and the pens out and before you know it you've gone down a long explanation cul de sac when all they wanted to do was get a [CID][] back.
```

File: content/post/104-js-ipfs-0.48.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+for await (const file of ipfs.addAll(files)) {
+  // All the files!
+}
+```
+
+### APIs with optional arguments
+
+Recently we [released a change](https://blog.ipfs.io/2020-05-21-js-ipfs-0-44/#cancellable-requests) that allowed passing [AbortSignal][]s to all API methods. This necessitated adding an `options` object to every API call that didn't have one already. This left us in the weird situation where some arguments were optional, but were not in the options argument. Worse, the actions of some API calls changed dramatically depending on whether you passed an option or not. For example `ipfs.bootstrap.rm([multiaddr])` would completely empty the bootstrap list if you didn't pass a [Multiaddr][].
+
+All this leads to weird behaviour and subtle bugs when you pass things like `undefined` in for an optional arg position and don't pass an options argument, as well as knotty, error-prone internal code that tries to guess what you passed where based on type or properties of the objects where their types are the same.
```
Comment (jacobheun) [inline content/post/104-js-ipfs-0.48.md:143]:
```suggestion
All this leads to weird behaviour and subtle bugs when you pass things like `undefined` in for an optional arg position and don't pass an options argument, as well as knotty, error-prone internal code that tries to guess what you passed based on type or properties of the objects where their types are the same.
```

File: content/post/104-js-ipfs-0.48.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+
+// after
+ipfs.bootstrap.add('/ip4/...') // adds a multiaddr to the bootstrap list
+ipfs.bootstrap.reset() // restores default bootstrappers
+```
+
+```javascript
+// before
+ipfs.bootstrap.rm('/ip4/...') // removes a multiaddr from the bootstrap list
+ipfs.bootstrap.rm({ all: true }) // restores default bootstrappers
```
Comment (jacobheun) [inline content/post/104-js-ipfs-0.48.md:162]:
```suggestion
ipfs.bootstrap.rm({ all: true }) // empties bootstrapper list
```

File: content/post/104-js-ipfs-0.48.md [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+
+There are several ways to give in-browser IPFS nodes a better experience on the network, one of those is [Delegate Nodes](https://blog.ipfs.io/2019-08-06-js-ipfs-0-37/#delegated-peer-and-content-routing). A Delegate Node is a network peer that performs certain actions on behalf of other nodes on the network. In this case it will make [DHT][] queries on our behalf so we can find more peers and more content than ever before.
+
+`js-IPFS@0.48.0` enables delegate nodes in the configuration by default. That means you should see far more peers that you have previously and be able to find content faster and more reliably than ever before.
+
+By default it uses [public delegate nodes](https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs/docs/DELEGATE_ROUTERS.md) to give you the best out-of-the-box experience. These nodes are a shared commons but have no availability guarentees and are a potentially a source of resource contention. If you are deploying JS IPFS in a production environment you should stand up your own delegate nodes and [configure JS IPFS](https://github.com/ipfs/js-ipfs/blob/master/docs/CONFIG.md#delegates) accordingly.
+
+## 🏓 DHT configuration
```
[the remark this answers is not in the document]
Comment (achingbrain) [reply content/post/104-js-ipfs-0.48.md:25]: AFAIK you can't run the node as a DHT server, it'll only ever be in client mode, but yes, will do - good to address expectations.

[the remark this answers is not in the document]
Comment (achingbrain) [reply content/post/104-js-ipfs-0.48.md:25]: I've added another paragraph about this.

**Pull request merged**